tempodb/encoding/common/config.go

```diff
@@ -8,17 +8,19 @@ import (
 
 // BlockConfig holds configuration options for newly created blocks
 type BlockConfig struct {
-	IndexDownsampleBytes int              `yaml:"index_downsample_bytes"`
-	IndexPageSizeBytes   int              `yaml:"index_page_size_bytes"`
-	BloomFP              float64          `yaml:"bloom_filter_false_positive"`
-	BloomShardSizeBytes  int              `yaml:"bloom_filter_shard_size_bytes"`
-	Version              string           `yaml:"version"`
-	Encoding             backend.Encoding `yaml:"encoding"`
-	SearchEncoding       backend.Encoding `yaml:"search_encoding"`
-	SearchPageSizeBytes  int              `yaml:"search_page_size_bytes"`
+	BloomFP             float64          `yaml:"bloom_filter_false_positive"`
+	BloomShardSizeBytes int              `yaml:"bloom_filter_shard_size_bytes"`
+	Version             string           `yaml:"version"`
+	SearchEncoding      backend.Encoding `yaml:"search_encoding"`
+	SearchPageSizeBytes int              `yaml:"search_page_size_bytes"`
+
+	// v2 fields
+	IndexDownsampleBytes int              `yaml:"v2_index_downsample_bytes"`
+	IndexPageSizeBytes   int              `yaml:"v2_index_page_size_bytes"`
+	Encoding             backend.Encoding `yaml:"v2_encoding"`
```

Comment:
We should warn if these fields are configured and version is not v2.

Reply:
+1 added